docs/source/production.rst

```diff
@@ -118,15 +118,18 @@ like this:
     hooks:
       - id: sqlfluff-lint
         # For dbt projects, this installs the dbt "extras":
-        # additional_dependencies: ['.[dbt]']
+        # additional_dependencies: ['sqlfluff-templater-dbt']
       - id: sqlfluff-fix
         # Arbitrary arguments to show an example
         # args: [--rules, "L003,L014"]
-        # additional_dependencies: ['.[dbt]']
+        # additional_dependencies: ['sqlfluff-templater-dbt']
 
 When trying to use the `dbt templater`_, uncomment the
 ``additional_dependencies`` to install the extras.
-This is equivalent to running ``pip install sqlfluff[dbt]``.
+This is equivalent to running ``pip install sqlfluff-templater-dbt``.
+
+Some dialects may require further ``additional_dependencies``.
```

Comment:
As far as I know, all dialects require adapters (https://docs.getdbt.com/docs/available-adapters) so perhaps this should be reworded.

I also think dbt-<adapter> dependency should come before sqlfluff-templater-dbt. That could be used to pin dbt to a specific version, as without that, sqlfluff-templater-dbt will pull in latest version of dbt-core

Reply:
Makes sense, I've updated the docs to refer the user to install the relevant adapter.

This is the original PR when this was split out: #1786. 
Looks like no documentation was added, so its worth also making a separate PR to update the <dbt-adapter> advice in https://docs.sqlfluff.com/en/stable/configuration.html#dbt-project-configuration as well.